src/Control/Carrier/Cull/Church.hs

```diff
@@ -53,6 +53,12 @@ instance Alternative (CullC m) where
       runReader cull (runCullC l) <|> runReader cull (runCullC r)
   {-# INLINE (<|>) #-}
 
+-- | Separate fixpoints are computed for each branch.
+--
+-- >>> run (runCullA @[] (take 3 <$> mfix (\ as -> pure (0 : map succ as) <|> pure (0 : map pred as))))
+-- [[0,1,2],[0,-1,-2]]
+deriving instance MonadFix m => MonadFix (CullC m)
```

Comment:
We’re still deriving this, but I wanted to demonstrate & document its behaviour.